plugins/importexport/native/filter/IssueGalleyNativeXmlFilter.inc.php

```diff
@@ -117,6 +119,28 @@ function addFile($doc, $issueGalleyNode, $issueGalley) {
 			$issueGalleyNode->appendChild($issueFileNode);
 		}
 	}
+
+	/**
+	 * Create and add identifier nodes to an issue galley node.
+	 * @param $doc DOMDocument
+	 * @param $issueGalleyNode DOMElement
+	 * @param $issueGalley IssueGalley
+	 */
+	function addIdentifiers($doc, $issueGalleyNode, $issueGalley) {
+		$deployment = $this->getDeployment();
+
+		// Add internal ID
+		$issueGalleyNode->appendChild($node = $doc->createElementNS($deployment->getNamespace(), 'id', $issueGalley->getId()));
+		$node->setAttribute('type', 'internal');
+		$node->setAttribute('advice', 'ignore');
+
+		// Add public ID
+		if ($pubId = $issueGalley->getStoredPubId('publisher-id')) {
+			$issueGalleyNode->appendChild($node = $doc->createElementNS($deployment->getNamespace(), 'id', htmlspecialchars($pubId, ENT_COMPAT, 'UTF-8')));
```

Comment:
I think it's cleaner (and probably safer) to use [`DOMElement::createTextNode`](https://www.php.net/manual/en/domdocument.createtextnode) rather than `htmlspecialchars` to do the escaping. (Here and probably elsewhere)